modules/apps/batch-planner/batch-planner-rest-impl/src/main/java/com/liferay/batch/planner/rest/internal/vulcan/batch/engine/FieldProvider.java

```diff
@@ -80,7 +82,25 @@ public List<Field> getFields(
 		Map<String, Field> fields = objectEntryOpenAPIResource.getFields(
 			uriInfo);
 
-		return new ArrayList<>(fields.values());
+		Map<String, Field> filteredFields = new HashMap<>();
+
+		for (Map.Entry<String, Field> entry : fields.entrySet()) {
+			Field field = entry.getValue();
+
+			if (Objects.equals(field.getType(), "object") ||
+				Objects.equals(field.getType(), "array")) {
+
+				String ref = field.getRef();
+
+				if ((ref != null) && ref.startsWith("#/components/schemas/"))
+
+					continue;
+			}
+
+			filteredFields.put(entry.getKey(), entry.getValue());
+		}
+
+		return new ArrayList<>(filteredFields.values());
```

Comment:
As the method returns an ArrayList, we could use the `TransformUtils.transform` method:

```
return TransformUtil.transform(
			fields.values(),
			field -> {
				if ((Objects.equals(field.getType(), "array") ||
					 Objects.equals(field.getType(), "object")) &&
					StringUtil.startsWith(
						field.getRef(), "#/components/schemas/")) {

					return null;
				}

				return field;
			});
```

What do you think? Maybe it is simpler